components/kyma-metrics-collector/pkg/edp/client.go

```diff
@@ -74,60 +73,62 @@ func (eClient Client) getEDPURL(dataTenant string) string {
 func (eClient Client) Send(req *http.Request, payload []byte) (*http.Response, error) {
 	var resp *http.Response
 	var err error
-	customBackoff := wait.Backoff{
-		Steps:    eClient.Config.EventRetry,
-		Duration: eClient.Config.Timeout,
-		Factor:   5.0,
-		Jitter:   0.1,
-	}
-	err = retry.OnError(customBackoff, func(err error) bool {
-		return err != nil
-	}, func() (err error) {
-		reqStartTime := time.Now()
-		// send request.
-		req.Body = io.NopCloser(bytes.NewReader(payload))
-		resp, err = eClient.HttpClient.Do(req)
-		duration := time.Since(reqStartTime)
-		// check result.
-		if err != nil {
-			urlErr := err.(*url.Error)
-			responseCode := http.StatusBadRequest
-			if urlErr.Timeout() {
-				responseCode = http.StatusRequestTimeout
+	defer func() {
+		if resp != nil && resp.Body != nil {
```

Comment:
according to the spec the body will never be nil, unless you get an error from the http.client

> If the returned error is nil, the Response will contain a non-nil Body which the user is expected to close. If the Body is not both read to EOF and closed, the Client's underlying RoundTripper (typically Transport) may not be able to re-use a persistent TCP connection to the server for a subsequent "keep-alive" request.


but you should defer the close after getting the response and checking it for error (line 94)

Reply:
refactored.